### PR TITLE
[stateless_validation] Enable distribution of state witness via PartialEncodedStateWitness

### PR DESCRIPTION
**Pull request by shreyan-gupta**
This PR adds a new protocol version to enable distribution of state witness via reed solomon encoding

Added integration tests and test loop implementation.

### CODE REVIEW

File: integration-tests/src/tests/client/features/orphan_chunk_state_witness.rs [hunk 1]
```diff
@@ -232,55 +231,6 @@ fn test_orphan_witness_valid() {
         .unwrap();
 }
 
-#[test]
-fn test_orphan_witness_bad_signature() {
```
Comment (shreyan-gupta) [inline integration-tests/src/tests/client/features/orphan_chunk_state_witness.rs:236]: These tests test signature which we are removing from state witness and pushing functionality to parts.

File: integration-tests/src/tests/client/features/multinode_test_loop_example.rs [hunk 2]
```diff
@@ -601,6 +603,32 @@ pub fn route_network_messages_to_client<
                     tracing::warn!("Dropping state-witness-ack message to self");
                 }
             }
+            NetworkRequests::PartialEncodedStateWitness(validator_witness_tuple) => {
```
Comment (robin-near) [inline integration-tests/src/tests/client/features/multinode_test_loop_example.rs:606]: fyi when you merge, this is now moved to client/.../test_loop.rs.

**Review by robin-near — APPROVED** [review]

File: chain/client/src/stateless_validation/partial_witness_tracker.rs [hunk 3]
```diff
@@ -26,19 +27,70 @@ const RATIO_DATA_PARTS: f32 = 0.8;
...
 
-    pub fn entry(&mut self, total_parts: usize) -> &ReedSolomon {
-        self.rs_map.entry(total_parts).or_insert_with(|| {
+    pub fn entry(&mut self, total_parts: usize) -> &ReedSolomonWrapper {
+        self.rs_map.entry(total_parts).or_insert_with(|| ReedSolomonWrapper::new(total_parts))
+    }
+}
+
+pub enum ReedSolomonWrapper {
+    SinglePart,
```
[the remark this answers is not in the document]
Comment (robin-near) [reply chain/client/src/stateless_validation/partial_witness_tracker.rs:44]: Would be nice to just document exactly that above the enum.

[the remark this answers is not in the document]
Comment (shreyan-gupta) [reply chain/client/src/stateless_validation/partial_witness_tracker.rs:44]: Sounds good! I'm planning to shift this behavior exception to within the cache entry as per [this comment](https://github.com/near/nearcore/pull/11154#discussion_r1583518949).

**Review by tayfunelmas — APPROVED** [review]

**Pull request merged**